### ISSUE TITLE
LibGUI: Don't run text change callbacks on every character when pasting or modifying large amounts of text at once

### ISSUE THREAD

**Issue by kleinesfilmroellchen** [SerenityOS/serenity#18608]
For a detailed discussion of this problem, including a suboptimal fix, see #17774 . The TL;DR is that since calling text change callbacks for every character of a large unified change is pointless, we should find a good way of inhibiting the callbacks while the change is going on. Some callbacks are very performance-intensive, some modify the document themselves, etc., so we want to call them as little as possible. I'm pretty sure that in the case of GMLPlayground, we should be able to achieve 2n runtime instead of the current n².

**Comment by ADKaster** [comment]
@kleinesfilmroellchen I don't think this one applies as a good first issue. Good first issues should be something that someone who is new to Github, new to open source, new to C++ and new to the project could pick up without a lengthy design discussion on discord.
